modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -329,25 +332,93 @@ fn generate_ephemeral_tweak(static_pk: PublicKey) -> ([u8; 32], PublicKey) {
 }
 
 impl LightningClientModule {
-    /// Updates the mapping from lightning node public keys to gateway api
-    /// endpoints maintained in the module database. When paying an invoice this
-    /// enables the client to select the gateway that has created the invoice,
-    /// if possible, such that the payment does not go over lightning, reducing
-    /// fees and latency.
-    ///
-    /// Client integrators are expected to call this function once a day.
-    pub async fn update_gateway_cache(&self) {
-        if let Ok(gateways) = self.module_api.gateways().await {
-            let mut dbtx = self.client_ctx.module_db().begin_transaction().await;
+    #[allow(clippy::too_many_arguments)]
+    fn new(
+        federation_id: FederationId,
+        cfg: LightningClientConfig,
+        notifier: ModuleNotifier<LightningClientStateMachines>,
+        client_ctx: ClientContext<Self>,
+        module_api: DynModuleApi,
+        keypair: KeyPair,
+        gateway_conn: Arc<dyn GatewayConnection + Send + Sync>,
+        admin_auth: Option<ApiAuth>,
+        task_group: &TaskGroup,
+    ) -> Self {
+        Self::spawn_gateway_map_update_task(
+            federation_id,
+            client_ctx.clone(),
+            module_api.clone(),
+            gateway_conn.clone(),
+            task_group,
+        );
+
+        Self {
+            federation_id,
+            cfg,
+            notifier,
+            client_ctx,
+            module_api,
+            keypair,
+            gateway_conn,
+            admin_auth,
+        }
+    }
+
+    fn spawn_gateway_map_update_task(
+        federation_id: FederationId,
+        client_ctx: ClientContext<Self>,
+        module_api: DynModuleApi,
+        gateway_conn: Arc<dyn GatewayConnection + Send + Sync>,
+        task_group: &TaskGroup,
+    ) {
+        task_group.spawn("gateway_map_update_task", move |handle| async move {
```

Comment:
@joschisan BTW. `spawn_cancellable` does the select and shutdown thing automatically.